openpos-server-core/src/main/java/org/jumpmind/pos/server/config/WebSocketConfig.java

```diff
@@ -77,6 +85,44 @@ public void configureMessageBroker(MessageBrokerRegistry config) {
     public void configureClientInboundChannel(ChannelRegistration registration) {
         registration.interceptors(new ExecutorChannelInterceptorAdapter() {
 
+            @Override
+            public Message<?> beforeHandle(Message<?> message, MessageChannel channel, MessageHandler handler) {
+                StompHeaderAccessor headerAccessor= StompHeaderAccessor.wrap(message);
+
+                if (StompCommand.CONNECT.equals(headerAccessor.getCommand())) {
+                    String sessionId = (String) message.getHeaders().get("simpSessionId");
+                    Map<String, List<String>> nativeHeaders = (Map<String, List<String>>) message.getHeaders().get("nativeHeaders");
+                    List<String> deviceToken = nativeHeaders.get("deviceToken");
+
+                    if (!deviceToSessionMap.containsKey(deviceToken.get(0))) {
+                        deviceToSessionMap.put(deviceToken.get(0), SessionContext.builder()
+                                .sessionId(sessionId)
+                                .build());
+                    }
+
+                    boolean isDeviceConnected = false;
+
+                    for (Map.Entry<String, SessionContext> pair : deviceToSessionMap.entrySet()) {
+                        if (pair.getValue().getSessionId().equals(sessionId)) {
+                            isDeviceConnected = true;
+                        }
+                    }
+
+                    if (!isDeviceConnected) {
+                        throw new DeviceNotAuthorizedException();
+                    }
```

Comment:
nit: you could early out this if you wanted to. Unlikely to save much because I imagine there won't be a high quantity of clients at any given time, but:

```suggestion
                    for (Map.Entry<String, SessionContext> pair : deviceToSessionMap.entrySet()) {
                        if (pair.getValue().getSessionId().equals(sessionId)) {
                            throw new DeviceNotAuthorizedException();
                        }
                    }
```